### PR TITLE
Rework "apt manually installed"

### PR DESCRIPTION
**Pull request by fkusei**
This should fix the concerns stated in https://github.com/bundlewrap/bundlewrap/pull/764 about being a performance regression.

### CODE REVIEW

**Comment by fkusei** [comment]
There seems to be another regression, because sometimes marking works, but the cache is not updated, possibly i'm doing something wrong here.

**Comment by fkusei** [comment]
Yes, i've held it wrong.

File: bundlewrap/items/pkg_apt.py [hunk 1]
```diff
@@ -14,8 +15,46 @@ class AptPkg(Pkg):
...
+                self.pkg_remove()
+            else:
+                self._pkg_manual_cache.setdefault(self.node.name, set()).add(pkg_name)
+                self.pkg_install()
+        elif 'mark' in status.keys_to_fix:
+            self._pkg_manual_cache.setdefault(self.node.name, set()).add(pkg_name)
+            self.run("apt-mark manual {}".format(quote(pkg_name)))
 
     def pkg_all_installed(self):
+        result = self.run("apt-mark showmanual")
```
Comment (vain) [inline bundlewrap/items/pkg_apt.py:53]: Populating the cache in this function (`pkg_all_installed()`) is racy, sadly. It’s not guaranteed that this function gets called *before* we query the cache for the first time.

Either use the pattern as `pkg_installed_cached()` / `pkg_all_installed()`, or maybe just move populating the cache into `pkg_manually_installed()`?

Comment (fkusei) [reply bundlewrap/items/pkg_apt.py]: In my tests, this worked every time. I have moved the function as requested nonetheless.

**Comment by vain** [comment]
Alrigthy, thanks! 👍

**Pull request merged**